src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -578,6 +576,43 @@ public static IEnumerable<object[]> CSharpInlineByteArrayTestData()
                                         "(new byte[s.Length]).AsMemory(0, (int)s.Length), new CancellationToken()" };
         }
 
+        [Fact]
+        public Task CS_Fixer_Diagnostic_EnsureSystemNamespaceAutoAdded()
+        {
+            string originalCode = @"
+using System.IO;
+using System.Threading;
+
+class C
+{
+    public async void M()
+    {
+        using (FileStream s = File.Open(""path.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[s.Length];
+            await s.ReadAsync(new byte[s.Length], 0, (int)s.Length);
+        }
+    }
+}";
+            string fixedCode = @"
+using System.IO;
+using System.Threading;
+using System;
```

Comment:
@carlossanlop I would have expected the code fix to have added the using in sorted order. Can you check the behavior when you apply the fix in VS with existing usings?

Reply:
@mavasani It does not get added at the beginning, I was also expecting that.
`generator.AddNamespaceImports` added it at the end. I verified this when debugging the new fixer code.

Reply:
Tagging @CyrusNajmabadi for insights here. How does code fixer ensure that a new using import that it wants to is sorted? Otherwise, I feel like you have a follow-up bug coming your way when someone complains the code fix broke the sort order of their imports.

Reply:
> Tagging @CyrusNajmabadi for insights here. How does code fixer ensure that a new using import that it wants to is sorted? 

The internal roslyn system for adding imports will add it in a manner that respects all the editor options here (sorting, spacing, etc.).